test/integration/copy_paste_spec.mjs

```diff
@@ -15,17 +15,18 @@
 
 import {
   closePages,
-  kbCopy,
+  copy,
   kbSelectAll,
   loadAndWait,
   mockClipboard,
   waitForEvent,
 } from "./test_utils.mjs";
 
 const selectAll = async page => {
-  const promise = waitForEvent(page, "selectionchange");
+  const generator = waitForEvent(page, "selectionchange");
```

Comment:
Can't we just add a third param to `waitForEvent` which will be here `async () => await kbSelectAll(page)` and then do the equivalent of the two `generator.next` in the `waitForEvent` ?
Or maybe keep `waitForEvent` like this and add another function to call the function triggering the event wrapped in generator.next calls,... something like that...

Reply:
Good idea! I was also hoping to keep most of that logic inside `waitForEvent`, but couldn't think of a better way to handle the control flow until you mentioned this, so I'll definitely give that a try; thanks!

Reply:
Fixed in the new commit. This is indeed much nicer/simpler!